src/server/components/annual-review/controller.ts

```diff
@@ -78,21 +87,47 @@ function formatDataForSummaryRows(listItem: ListItemGetObject): Types.govukRow[]
   return mergedRows;
 }
 
-export function confirmPostController(req: Request, res: Response): void {
-  const chosenValue = req.body["is-your-information-correct"];
+export async function confirmPostController(req: Request, res: Response, next: NextFunction): Promise<void> {
+  try {
+    const chosenValue = req.body["is-your-information-correct"];
 
-  if (!chosenValue) {
-    req.flash("annualReviewError", "Select if your information is correct or if you need to update it");
-    return res.redirect(`/annual-review/confirm/${req.body.reference}`);
-  }
+    if (!chosenValue) {
+      req.flash("annualReviewError", "Select if your information is correct or if you need to update it");
+      return res.redirect(`/annual-review/confirm/${req.body.reference}`);
+    }
+
+    if (chosenValue === "yes") {
+      return res.redirect(`/annual-review/declaration/${req.body.reference}`);
+    }
 
-  if (chosenValue === "yes") {
-    return res.redirect(`/annual-review/declaration/${req.body.reference}`);
+    if (chosenValue === "no") {
+      await redirectToFormRunner(req, res, next);
```

Comment:
probably needs another check for req.params.listItemRef here to ensure the ref is provided before calling the DB in redirectToFormRunner()

Reply:
it's fine as it is on a few counts 

1. this route is `/annual-review/confirm/:listItemRef`. There is no route that is just `/annual-review/confirm`. So if :listItemRef is undefined, you just get a 404. 
2. if somehow `redirectToFormRunner` is called with an undefined :listItemRef, `findListItemByReference` will return null. 
3. if nothing is found (i.e. null returned) `  if (!result) {
    return next(new HttpException(404, "404", "The list item cannot be found"));
  }`

Reply:
makes sense.